extensions/src/test/java/org/pentaho/platform/plugin/services/security/userrole/oauth/PentahoOAuthLoginAuthenticationFilterTest.java

```diff
@@ -0,0 +1,207 @@
+    DefaultOidcUser defaultOidcUser = mock( DefaultOidcUser.class );
+    Authentication authentication = mock( OAuth2AuthenticationToken.class );
+    when( ( (OAuth2AuthenticationToken) authentication ).getAuthorizedClientRegistrationId() ).thenReturn( "test" );
+    when( authentication.getPrincipal() ).thenReturn( defaultOidcUser );
+    when( defaultOidcUser.getAttribute( "roles" ) ).thenReturn( List.of( "Administrator" ) );
+    when( roleMapper.toPentahoRole( "Administrator" ) ).thenReturn( "Administrator" );
+
+    Collection<GrantedAuthority> authorities = List.of( new SimpleGrantedAuthority( "Administrator" ) );
+    doReturn( authorities ).when( authentication ).getAuthorities();
+
+    try (
+      MockedStatic<PentahoSessionHolder> pentahoSessionHolderMockedStatic = mockStatic( PentahoSessionHolder.class );
+      MockedStatic<PentahoSystem> mockedPentahoSystemMockedStatic = mockStatic( PentahoSystem.class ) ) {
+      pentahoSessionHolderMockedStatic.when( PentahoSessionHolder::getSession )
+        .thenReturn( mock( IPentahoSession.class ) );
+      mockedPentahoSystemMockedStatic.when(
+          () -> PentahoSystem.get( String.class, "singleTenantAuthenticatedAuthorityName", null ) )
+        .thenReturn( "Authenticated" );
+      PentahoOAuthLoginAuthenticationFilter filter =
+        new PentahoOAuthLoginAuthenticationFilter( clientRegistrationRepository, authorizedClientService, roleMapper );
+      assertNotNull( filter.getOAuth2AuthenticationToken( authentication ) );
+    }
+  }
+
+  @Test
+  public void testGetOAuth2AuthenticationTokenNoRolesAndAuthorities() {
+    DefaultOidcUser defaultOidcUser = mock( DefaultOidcUser.class );
+    Authentication authentication = mock( OAuth2AuthenticationToken.class );
+    when( ( (OAuth2AuthenticationToken) authentication ).getAuthorizedClientRegistrationId() ).thenReturn( "test" );
+    when( authentication.getPrincipal() ).thenReturn( defaultOidcUser );
+    when( defaultOidcUser.getAttribute( "roles" ) ).thenReturn( Collections.emptyList() );
+    when( authentication.getAuthorities() ).thenReturn( Collections.emptyList() );
+
+    try (
+      MockedStatic<PentahoSessionHolder> pentahoSessionHolderMockedStatic = mockStatic( PentahoSessionHolder.class );
+      MockedStatic<PentahoSystem> mockedPentahoSystemMockedStatic = mockStatic( PentahoSystem.class ) ) {
+      pentahoSessionHolderMockedStatic.when( PentahoSessionHolder::getSession )
+        .thenReturn( mock( IPentahoSession.class ) );
+      mockedPentahoSystemMockedStatic.when(
+          () -> PentahoSystem.get( String.class, "singleTenantAuthenticatedAuthorityName", null ) )
+        .thenReturn( "Authenticated" );
+      PentahoOAuthLoginAuthenticationFilter filter =
+        new PentahoOAuthLoginAuthenticationFilter( clientRegistrationRepository, authorizedClientService, roleMapper );
+      assertNotNull( filter.getOAuth2AuthenticationToken( authentication ) );
+    }
+  }
+
+}
```

Comment:
Alot of your files don't have an newline at the end of the file, the should violate a our check style too.

Reply:
addressed